test/AutoDiff/compiler_crashers_fixed/sr12732-optimize-partial-apply-convention-thin-only.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %target-build-swift -Osize %s
+// REQUIRES: asserts
+
+// SR-12732: Fix `partial_apply` optimization.
+
+// Do not rewrite `partial_apply` to `thin_to_thick_function` if the specialized
+// callee is not `@convention(thin)`.
+
+import DifferentiationUnittest
```

Comment:
It would be nice to remove the dependency on `DifferentiationUnittest`. I tried but didn't quite figure it out: `-enable-library-evolution` seems involved, but passing it exposes another bug (TF-1274).

Let's visit this later to unblock progress.